src/tss2-esys/esys_crypto_ossl.c

```diff
@@ -631,10 +629,14 @@ iesys_cryptossl_pk_encrypt(TPM2B_PUBLIC * pub_tpm_key,
                    "Could not create evp key.", cleanup);
     }
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
-    rsa_key->n = BN_bin2bn(pub_tpm_key->publicArea.unique.rsa.buffer,
+    if (!BN_bin2bn(pub_tpm_key->publicArea.unique.rsa.buffer,
                            pub_tpm_key->publicArea.unique.rsa.size,
-                           NULL);
+                           rsa_key->n)) {
+        goto_error(r, TSS2_ESYS_RC_GENERAL_FAILURE,
+                   "Could not create rsa n.", cleanup);
+    }
 #else
+    BIGNUM *n = NULL;
     if (!(n = BN_bin2bn(pub_tpm_key->publicArea.unique.rsa.buffer,
                         pub_tpm_key->publicArea.unique.rsa.size,
                         NULL))) {
```

Comment:
FYI I thought about using RSA_get0_key() to get a pointer to N and then passing that to BN_bin2bn() to help further reduce the delta between OSSL versions, but RSA_get0_key() returns a const pointer, they don't want you doing things like what Initially thought.

Reply:
We could probably use RSA_set_method() and provide our own esys_rsa_keygen() to make it look the same on both ossl < 1.1.0 and ossl >=1.1.0.

Reply:
That's kind of how we did it in tools... just added the missing functions